File: Chapter 06/C Code/_6_9.c
#include<stdio.h>
#include<stdlib.h>

int main () {
  // variable to store the result
  long bina = 0;
  int re, i = 1;
  int nn, bb;
  printf("Enter the decimal number: ");
  scanf("%d", &nn);

  printf("Enter the required base: ");
  scanf("%d", &bb);

  if(bb<2 || bb > 9)
  {
      printf("Base should be between 2 to 9");
      exit(0);
  }
 // loop to convert to binary
  while (nn != 0) {
        // get remainder of nn divided by base
    re = nn % bb;
        // divide nn by base
    nn = nn/bb;
        // multiply remainder by i
    // add the product to bin
    bina += re * i;
        // multiply i by 10
    i *= 10;
  }

  printf("The base %d representation of %d is %ld", bb,nn,bina);

return 0;
}

File: Chapter 06/Python Code/_6_9.py
bina = 0
i = 1
nn=int(input("Enter the decimal number: "))
bb=int(input("Enter the required base: "))

if bb<2 or bb > 9:
    print("Base should be between 2 to 9")
    exit(0)
    
while nn != 0 : 
    re = nn % bb
    nn = nn//bb
    bina += re * i
    i *= 10
  
print("The base ", bb, " representation of ", nn, " is ", bina)
